c10/cuda/CUDADeviceAssertion.h

```diff
@@ -0,0 +1,104 @@
+#pragma once
+
+#include <c10/cuda/CUDAException.h>
+#include <c10/macros/Macros.h>
+
+namespace c10 {
+namespace cuda {
+
+#ifdef TORCH_USE_CUDA_DSA
+// Copy string from `src` to `dst`
+static __device__ void dstrcpy(char* const dst, const char* const src) {
```

Comment:
Not sure if it supported across all cuda versions, but isn't it just `strncpy`? (and if it is not, perhaps it should be called `strncpy`

Also, not sure if `static` keyword has any effect on `__device__` only functions (as they do not have global visibility to beging with)

Reply:
Leaving the `static` since my recollection is someone else requested it earlier in review and it seems otherwise benign.

I don't think `strncpy` works in CUDA and am not sure if we can adequately test for that, so I'd prefer to keep this for now.

This function's signature differs from `strncpy`, so I'd prefer to keep the name as-is unless you have other suggestions.

tl;dr I haven't made a change here.